app/src/main/java/fr/neamar/kiss/MainActivity.java

```diff
@@ -181,6 +183,7 @@ public void onCreate(Bundle savedInstanceState) {
          * Initialize all forwarders
          */
         forwarderManager = new ForwarderManager(this);
+        permissionManager = new Permission(this);
```

Comment:
Let's only construct if we need? It's not super common.

Reply:
Unfortunately, as discussed, we need Activity to request permissions, and we possibly need to request permissions at any time, including a current usage in ContactProvider, and there is no access to MainActivity there, thus lazy init is currently difficult to implement.